src/main.c

```diff
@@ -87,6 +87,51 @@ void draw_testfb()
 	}
 }
 
+void poweroff()
+{
+	// Stop wasting energy
+	GPIOA_ModeCfg(GPIO_Pin_All, GPIO_ModeIN_Floating);
+	GPIOB_ModeCfg(GPIO_Pin_All, GPIO_ModeIN_Floating);
+
+	// Configure wake-up
+	GPIOA_ModeCfg(KEY1_PIN, GPIO_ModeIN_PD);
+	GPIOA_ITModeCfg(KEY1_PIN, GPIO_ITMode_RiseEdge);
+	PFIC_EnableIRQ(GPIO_A_IRQn);
+	PWR_PeriphWakeUpCfg(ENABLE, RB_SLP_GPIO_WAKE, Long_Delay);
+
+	/* Good bye */
+	LowPower_Shutdown(0);
+}
+
+void handle_mode_transition()
+{
+	static int prev_mode;
+	if (prev_mode == mode) return;
+
+	switch (mode)
+	{
+	case DOWNLOAD:
+		// Disable fb transition while in download mode
+		btn_onOnePress(KEY2, NULL);
+
+		// Take control of the current fb to display 
+		// the Bluetooth animation
+		while (mode == DOWNLOAD) {
+			/* Animation and Bluetooth will be placed here */
+		}
+		// If not being flashed, pressing KEY1 again will 
+		// make the badge goes off:

```

Comment:
fallsthrough, missing break

Reply:
No need for break as `mode` will be `POWER_OFF` once `change_mode()` get called in button handler. Additionally, if frames are downloaded, the badge will get reset itself.

Reply:
as this is unlabeled, compiler should scream warnings about this

Reply:
My toolchain (MRS_Toolchain_Linux_x64_V1.91, riscv-none-embed-gcc-8.2.0) was not turn on -Wfallthrough by default.

Should I turn on -Wextra and add `__attribute__ ((fallthrough))`?

Reply:
warnings are comment aware.

// fallthrough